src/Runner.Listener/Configuration/ServiceControlManager.cs

```diff
@@ -49,12 +49,12 @@ public void CalculateServiceName(RunnerSettings settings, string serviceNamePatt
 
             serviceName = StringUtil.Format(serviceNamePattern, repoOrOrgName, settings.AgentName);
 
-            if (serviceName.Length > 80)
```

Comment:
should we always add some random chars once we start trimming names?
We don't know how customers are going to name their repository and their runner, so we don't have confidence about the service name will still be unique on the machine even after we trimmed X from NWO and Y from the runner name.

My idea is always to add 4 random chars to the end once we trim anything.

Reply:
Yeah, I'll add a 4 digit number!